HIP/hip-0000-token-update-remove-keys.md

```diff
@@ -0,0 +1,196 @@
+
+## Security Implications
+
+Generally with administrative keys there are security requirements about how to secure and manage these secrets. This becomes increasingly important with this change, as a potential attacker could gain access to the admin keys and subsequently remove them from the entity - however, this would effectively lock/freeze them out, as it would the original administrator. These security considerations are not unique to this proposal and generally consistent with all keys attached to entities within the Hedera network.
+
+## How to Teach This
+
+The documentation for the [Token Service - Token Update](https://docs.hedera.com/guides/docs/sdks/tokens/update-a-token) would be updated to add examples on how to remove keys from a Token.
+
+## Reference Implementation
+
+## Rejected Ideas
+
+There was a discussion to introduce a dedicated method for removing keys. It opened up too many edge cases and since the solution proposed above solves the problem, it was decided to go with that approach.
+
+Here was the second option:
+
+### Option 2. Dedicated Remove Key Action
+
+An alternative approach might be to have a dedicated transaction `RemoveKeysTransaction`.
+
+This would take 3 parameters
+`tokenId` - The token to update
+`removeKey` - An enum representing a key Key.Wipe, Key.Freeze etc
+`removeKeys` - An array of Key enums [Key.Pause, Key.FeeSchedule, Key.Admin]
+
+**An example of removing the wipe key from an NFT**
+
+```js
+const transaction = new RemoveKeysTransaction()
+  .setTokenId(tokenId)
+  .setRemoveKey(Key.Wipe)
+  .freezeWithClient(client);
+
+// Sign the transaction with the admin key
+const signTx = await transaction.sign(adminKey);
+```
+
+**An example of removing multiple keys from a Token**
+
+```js
+const transaction = new RemoveKeysTransaction()
+  .setTokenId(tokenId)
+  .setRemoveKeys([Key.Pause, Key.FeeSchedule, Key.Kyc])
+  .freezeWithClient(client);
+
+// Sign the transaction with the admin key
+const signPauseTx = await transaction.sign(adminKey);
+```
+
+## Open Issues
+
+Greg Scullard raises an important point on why an early decision was made to prevent the removal of keys. This should be discussed as part of this HIP review so everyone is clear how removing certain keys could affect the token.
+
+> Generally speaking, whenever a key is set on an entity, it cannot be removed (so the same applies to supply, freeze, etc... keys). This is an early design decision which makes sense for some keys, if you had a freeze key and frozen accounts, unsetting the key would mean these accounts would be frozen for ever, unless a check is made which could be costly in terms of performance… - Greg Scullard [Discord](https://discord.com/channels/373889138199494658/616725732650909710/935199340555800616)
```

Comment:
Thank you for including this open issue in the HIP. I think we should add to the HIP a list of each type of key, and what the expected behavior is if the key is removed from each permutation of state that affects the key. The edge cases are the thing that concerns me most. 

I have a token. The token was frozen with the freeze key. The token was going to be frozen for only a short time until burned, but before it was burned the admin removed the burn and freeze keys. Now, the token is stuck in my account and I can **never** get rid of it. I will be stuck with this token forever.

We need to work through all these edge cases.

Reply:
@michielmulders let's work thru these edge cases together. 

On a slightly different note, i'm still a fan of being able to send any token back to the treasury as a type of "burn" mechanism. That way, nobody will be stuck with a token that they don't want and cannot get rid of. Altho I think that would be best in its own HIP.

Reply:
Agree, should be able to return to treasury at all times.